### PR TITLE
Adding shadow DOM support.

### PR DESCRIPTION
**Pull request by exsurgo**
Adding shadow DOM option.
* Adding `options` parameter to allow for further enhancements
* Adding `options.shadow` option, which creating and attaches shadow DOM, rather than light DOM
* Adding additional docs for shadow DOM option.

### CODE REVIEW

**Comment by justinbmeyer** [comment]
Awesome!

File: react-to-webcomponent.js [hunk 1]
```diff
@@ -20,12 +20,22 @@ var define = {
...
+
+/**
+ * Converts a React component into a webcomponent by wrapping it in a Proxy object.
+ * @param {ReactComponent}
+ * @param {React}
+ * @param {ReactDOM}
+ * @param {Object} options - Optional parameters
+ * @param {String?} options.shadow - Use shadow DOM rather than light DOM.
+ */
+export default function(ReactComponent, React, ReactDOM, options= {}) {
```
Comment (justinbmeyer) [inline react-to-webcomponent.js:32]: Is `options = {}` valid in every browser that supports Proxy?  If not, we might need to write this out "old school".  I don't think we are currently transpiling this file.

Comment (bmomberger-bitovi) [reply react-to-webcomponent.js:32]: Yes.
The only known* difference according to caniuse is Edge 12 and 13 supporting default function parameters and _not_ supporting Proxy.
https://caniuse.com/mdn-javascript_functions_default_parameters
https://caniuse.com/proxy

| * (some browsers are listed in the source docs with unknown feature sets)

**Comment by cherifGsoul** [comment]
This closed for #9 , thank you @exsurgo !

**Pull request closed without merging**